### PR TITLE
topdown/token: use BuiltinContext's time if no time constraint provided 

### LINKED ISSUE #3105: io.jwt.decode_verify should use time from built-in context instead of calling time.Now
If the `io.jwt.decode_verify` caller doesn't provide `time` member in the parameters, the built-in function implementation calls `time.Now()` to obtain the wallclock time for comparison against the `exp` and `nbf` claims:

- https://github.com/open-policy-agent/opa/blob/cc1cecf567d8df37787fd7b44e7e73fa7e0baa0c/topdown/tokens.go#L998
- https://github.com/open-policy-agent/opa/blob/cc1cecf567d8df37787fd7b44e7e73fa7e0baa0c/topdown/tokens.go#L1011

We should modify the built-in function to use the [time field on the built-in context](https://pkg.go.dev/github.com/open-policy-agent/opa@v0.26.0/topdown#BuiltinContext) as this allows evaluator to control the wallclock time using the [`rego.Time` or `rego.EvalTime`](https://pkg.go.dev/github.com/open-policy-agent/opa@v0.26.0/rego#Time) options.

### PR DESCRIPTION
**Pull request by srenatus**
Also cleans up the testdata a bit: 
- removing the data that isn't used;
- removing the call to time.now_ns() where it doesn't add anything.

Fixes #3105.

### CODE REVIEW

**Comment by srenatus** [comment]
ℹ️ Reviewers: it helps to only display `.go` file changes; the yaml changes are immaterial.

File: topdown/tokens_test.go [hunk 1]
```diff
@@ -117,19 +112,41 @@ OHoCIHmNX37JOqTcTzGn2u9+c8NlnvZ0uDvsd1BmKPaUmjmm
...
+			c := ast.NewObject()
+			c.Insert(ast.StringTerm("time"), wallclock)
+			constraints, err := parseTokenConstraints(c, ast.NumberTerm(json.Number("12134")))
+			if err != nil {
+				t.Fatalf("parseTokenConstraints: %v", err)
+			}
+			diff := now.UnixNano() - constraints.time
+			if diff > 1000 {
+				t.Errorf("expected time constraint to be within 1000ns of the one passed in, got %dns", diff)
+			}
```
Comment (tsandall) [inline topdown/tokens_test.go:130]: Why do we need this subtraction? Can't we check if now.UnixNano() is equal to constraints.time? I would have expected the values to be exactly the same. Do we need to use `time.Now` at all?

File: topdown/tokens_test.go [hunk 2]
```diff
@@ -117,19 +112,41 @@ OHoCIHmNX37JOqTcTzGn2u9+c8NlnvZ0uDvsd1BmKPaUmjmm
...
+		t.Run("unset, defaults to wallclock", func(t *testing.T) {
+			c := ast.NewObject() // 'time' constraint is unset
+			constraints, err := parseTokenConstraints(c, wallclock)
+			if err != nil {
+				t.Fatalf("parseTokenConstraints: %v", err)
+			}
+			diff := now.UnixNano() - constraints.time
+			if diff > 1000 {
+				t.Errorf("expected time constraint to be within 1000ns of the one passed as default, got %dns", diff)
+			}
```
Comment (tsandall) [inline topdown/tokens_test.go:142]: Similar to the above, I thought this was testing that one time is chosen over the other. Do we need subtraction and the actual `time.Now` value?

**Review by tsandall — COMMENTED** [review]
Left a couple comments on the new tests--mostly just for my own understanding. Other than that, LGTM. Thanks for cleaning up the implementation!

File: topdown/tokens_test.go (hunk 1 above)
Comment (srenatus) [reply topdown/tokens_test.go:130]: > I would have expected the values to be exactly the same

So would I. Anyways, there was a difference. Turns out we can just put some string "123" into it, and get it back here, and everything will be exact; the situation only differs, by some diff < 1ms, in the time.Now() case. I'll have a closer look once more!

**Comment by srenatus** [comment]
When using the stdlib, and fiddling with the time representations, test cases that are supposed to be "equal" (i.e. "exp is equal to passed time") *aren't anymore*. Trying to figure out what's correct about this and what isn't.

**Comment by srenatus** [comment]
An `int64` that should have been a `float64`. Imprecision fixed ✔️

**Review by tsandall — APPROVED** [review]
LGTM

**Pull request merged**